qiskit/aqua/algorithms/minimum_eigen_solvers/qaoa/qaoa.py

```diff
@@ -129,6 +132,7 @@ def __init__(self,
                          None,
                          optimizer,
                          initial_point=initial_point,
+                         gradient=gradient,
                          expectation=expectation,
                          include_custom=include_custom,
                          max_evals_grouped=max_evals_grouped,
```

Comment:
You are right! 
The max_evals grouping seems to happen in the finite difference method.
Can we write a method
`    @staticmethod
    def wrap_grad_function(function, args):`
to take care of this.
This would give the user more flexibility for the optimisation routine.